projects/core/src/main/resources/data/computercraft/lua/rom/programs/shell.lua

```diff
@@ -54,6 +54,67 @@ else
     bgColour = colours.black
 end
 
+local function tokenise(...)
+    local sLine = table.concat({ ... }, " ")
+    local tWords = {}
+    local bQuoted = false
+    for match in string.gmatch(sLine .. "\"", "(.-)\"") do
+        if bQuoted then
+            table.insert(tWords, match)
+        else
+            for m in string.gmatch(match, "[^ \t]+") do
+                table.insert(tWords, m)
+            end
+        end
+        bQuoted = not bQuoted
+    end
+    return tWords
+end
+
+-- Execute a program using os.run, unless a shebang is present.
+-- In that case, execute the program using the interpreter specified in the shebang.
+-- This may occur recursively, up to the maximum number of times specified by remainingRecursion
+-- Returns the same type as os.run, which is a boolean indicating whether the program exited successfully.
+local function executeProgram(remainingRecursion, env, path, ...)
+    local file = fs.open(path, "r")
+    if not file then
+        printError("File not found: " .. path)
+        return false
+    end
```

Comment:
```suggestion
    local file, err = fs.open(path, "r")
    if not file then
        printError(err)
        return false
    end
```

I wonder if it'd be better to do this instead? I don't know.